docs/news/posts/2024-10-04-data-access.md

```diff
@@ -0,0 +1,226 @@
+executable of Quantum ESPRESSO:
+
+![PwCalculation dump](./_gifs/calculation-dump-white-10fps-2160p.gif)
+
+And for a more complex `PwBandsWorkChain` [^3]:
+
+![PwBandsWorkChain dump](./_gifs/workflow-dump-white-10fps-2160p.gif)
+
+As you can see, the command works for both, individual calculations, as well as nested workflows [^4], resulting in
+the following output directories:
+
+**`tree` on a dumped example `CalcJob`**
+
+```shell
+dump-PwCalculation-54
+├── README.md
+├── inputs
+│  ├── _aiidasubmit.sh
+│  └── aiida.in
+├── outputs
+│  ├── _scheduler-stderr.txt
+│  ├── _scheduler-stdout.txt
+│  ├── aiida.out
+│  └── data-file-schema.xml
+└── node_inputs
+   └── pseudos
+      └── Si
+         └── Si.pbesol-n-rrkjus_psl.1.0.0.UPF
+```
+
+**`tree -d ` on a dumped example `WorkChain`**
+
+```shell
+dump-PwBandsWorkChain-70
+├── 01-relax-PwRelaxWorkChain
+│  ├── 01-iteration_01-PwBaseWorkChain
+│  │  ├── 01-create_kpoints_from_distance
+│  │  │  └── inputs
+│  │  └── 02-iteration_01-PwCalculation
+│  │     ├── inputs
+│  │     ├── node_inputs
+│  │     │  └── pseudos
+│  │     │     └── Si
+│  │     └── outputs
+│  └── 02-iteration_02-PwBaseWorkChain
+│     ├── 01-create_kpoints_from_distance
+│     │  └── inputs
+│     └── 02-iteration_01-PwCalculation
+│        ├── inputs
+│        ├── node_inputs
+│        │  └── pseudos
+│        │     └── Si
+│        └── outputs
+├── 02-seekpath-seekpath_structure_analysis
+│  └── inputs
+├── 03-scf-PwBaseWorkChain
+│  ├── ...
+...
+```
+
+Therefore, after running the command once, you'll have all data involved in the execution of your workflow directly
+accessible, and can explore it with the Unix command line utilities we all know and love [^5].
+
+So happy grepping!
+
+## New QueryBuilder Syntax
+
+_SQL queries, but intuitive!_
+
+In addition to accessing raw files as outlined above, AiiDA's powerful SQL database allows querying for stored
+`Node`s, which can be achieved with the `QueryBuilder` (as documented
+[here](https://aiida.readthedocs.io/projects/aiida-core/en/v2.6.2/howto/query.html)). While using the `QueryBuilder` is
+(at least for most of us) easier than writing raw SQL queries, its syntax usually requires some familiarization [^6].
+
+Recent improvements have therefore enabled an alternative, more intuitive way to construct queries. Let us explain with
+the following example: Assume you wanted to obtain all integers with `pk`s in a range between 1 and 10 (both excluded)
+from a `Group` called "integers". You'd have to construct the following, rather convoluted query:
+
+```python
+from aiida import orm
+
+qb = orm.QueryBuilder()
+qb.append(
+    orm.Group,
+    filters={
+        "label": "integers",
+    },
+    project=["label"],
+    tag="group",
+)
+qb.append(
+    orm.Int,
+    with_incoming="group",
+    filters={"and": [{"pk": {">": 1}}, {"pk": {"<": 10}}]},
+    project=["pk", "attributes.value"],
+)
+```
```

Comment:
Since not all might immediately understand what is going on, maybe you can briefly explain it? E.g. "after importing the needed modules in lines 1-2, we create a QueryBuilder object in line 4. In line 5-12 we filter for a group..."